pymodbus/repl/client/main.py

```diff
@@ -146,19 +146,19 @@ def _process_args(args, string=True):
                 if not string:
                     if "," in val:
                         val = val.split(",")
-                        val = [int(v) for v in val]
+                        val = [int(v, 0) for v in val]
```

Comment:
How is this helping?

From the [docs](https://docs.python.org/3/library/functions.html#int) `Base 0 means to interpret exactly as a code literal, so that the actual base is 2, 8, 10, or 16, and so that int('010', 0) is not legal, while int('010') is, as well as int('010', 8).`

Reply:
Can you share some outputs with out this change and with this change please?

Reply:
I did a short review of your code, and I agree with @dhoomakethu it do look strange. Your idea is OK, but you need to work on the implementation (and solve your rebase problem).

Reply:
int(<str>, 0) takes any sensible value as numerical. The original request was for accepting hex values. int(<str>, 0) accepts any value, with the cost that int('010', 0) throws ValueException - that can be handled by catching it and trying int(<str>), perhaps in a function, as this is on multiple places in the code.

I am not sure about rebasing issues. I discovered problems in my previous PR (#1041), which I fixed with this. Do you want to make an extra PR for improving those?

Reply:
As I already told you, your problems come from the fact that you use dev to submit PRs. You need to submit a PR based on the latest dev, whether you do it in this PR or open a new PR is up to you. I gave you a recipe how to work with dev and branches for each PR.

You need to prove that your PR accept hex values without causing problems. I believe that '010' is accepted today (without having tested it), so it needs to be accepted in this PR as well.

The best way to show your changes works as expected, is to write some test cases in test/test_<something>.py

Reply:
Ok. Let's close this one, I will make two new PRs, one for fixing problems created by #1041, second to address #628, with still accepting 010 and providing tests to show that.

Reply:
I was not aware that there are any problems due to #1041, I suggest you solve your dev problem first. If there are problems I look forward to see a PR.

#628 is a long standing issue, so it will be nice to have that fixed.